File: Test/practice/employee.cpp
#include "employee.h"

int Employee::getEmpId() const
{
   return empId;
}
string Employee::getName() const
{
   return name;
}
double Employee::getSalary() const
{
   return salary;
}

void Employee::Accept()
{
   cout << "Enter Employee details" << endl;
   cout << "Enter EmpID : ";
   cin >> empId;
   cout << "Enter Employee name : ";
   cin >> name;
   cout << "Enter Salary : ";
   cin >> salary;
}

void Employee::Display()
{
   cout << "Details of the employee are : " << endl;
   cout << "Employee id : " << empId << " name : " << name << " Salary : " << salary << endl;
}

bool Employee::sortEmp(const Employee &e1, const Employee &e2)
{
   if (e1.empId != e2.empId)
   {
      return e1.empId < e2.empId;
   }
   else if (e1.salary != e2.salary)
   {
      return e1.salary < e2.salary;
   }
   else
   {
      return e1.name < e2.name;
   }
}

File: TejasWorkplace/DayWisePrograms/src/day3/Employee.java
package day3;

public class Employee {
	private static long nextEmpNo = 123456;
	private final long empNo;
	private String name;
	private double salary; 

	public Employee() {
		this.empNo = nextEmpNo++;
		this.name = "";
		this.salary = 0.0;
	}

	public Employee(String name, double salary) {
		this.empNo = nextEmpNo++;
		this.name = name;
		this.salary = salary;
	}

	public long getEmpNo() {
		return empNo;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public double getSalary() {
		return salary;
	}

	public void setSalary(double salary) {
		this.salary = salary;
	}
	
	public String printData() {
		return "Employee [ Employee No. : " + empNo + ", Name : "+name+", salary : " + salary + " ]"; 
	}

}
